### ISSUE TITLE
Enhancement: Parler-TTS, a new open source model with steerable voice characteristics

### ISSUE THREAD

**Issue by phirsch** [mkiol/dsnote#122]
Wondering whether this could be an interesting model to add - it allows the voice characteristics to be altered using an auxiliary prompt.

Quote from https://github.com/huggingface/parler-tts:

> Parler-TTS is a lightweight text-to-speech (TTS) model that can generate high-quality, natural sounding speech in the style of a given speaker (gender, pitch, speaking style, etc). It is a reproduction of work from the paper [Natural language guidance of high-fidelity text-to-speech with synthetic annotations](https://www.text-description-to-speech.com/) by Dan Lyth and Simon King, from Stability AI and Edinburgh University respectively.
> Contrarily to other TTS models, Parler-TTS is a fully open-source release. All of the datasets, pre-processing, training code and weights are released publicly under permissive license, enabling the community to build on our work and develop their own powerful TTS models.

P.S.: Thanks for this great software - and extra kudos for the modular GPU acceleration FlatPak release!

**Comment by mkiol** [comment]
Thanks for letting know about this model. This creation of voice characteristics using text prompt is very interesting and unique! Parler-TTS would be a great addition to Speech Note.

The only problem is that this model cannot generate long text. Therefore, you need split the text into sentences and generate each sentence separately. Unfortunately, this causes each sentence to be generated in a different voice because [this problem](https://github.com/huggingface/parler-tts/issues/11) occurs.

I'm waiting for Parler-TTS devs to fix this problem and then I will definitely integrate it into Speech Note.

**Comment by phirsch** [comment]
Valid points there - thanks for considering the model and finding this out!

I agree that switching voices between any two consecutive sentences is pretty awkward (IIRC, some included models - maybe WhisperSpeech? - already exhibit this to some extent) - particularly for my main use case (helping to quickly but still accurately skim scientific papers).

Feel free to close the issue unless you prefer to keep it open as a reminder.

**Comment by mkiol** [comment]
Just released [v4.8.0 Beta 2](https://github.com/mkiol/dsnote/issues/223) with the following:

- Text to Speech
  - [_Parler-TTS_](https://github.com/huggingface/parler-tts) engine. Parler-TTS can generate high-quality, natural sounding speech in the style of a given speaker (gender, pitch, speaking style, etc). The speaker's characteristics are defined by a text description (prompt). To use Parler-TTS models, you need to configure a _Text voice profile_. This can be done in the _Voice profiles_ menu. Parler-TTS primarily supports English, but a multilingual model for French, Spanish, Portuguese, Polish, German, Dutch and Italian is also included. Currently, the multilingual model provides rather poor quality and not entirely usable speech. Parler-TTS works best with CUDA acceleration. CPU only processing can be very slow.

The consistency of voice in Parler-TTS is still not perfect, but it is usable.

To define "voice characteristics", use _Voice Profile_ menu, and select _Text voice profile_.

![Image](https://github.com/user-attachments/assets/3c328349-3ac0-4fa9-b10c-83718728169a)
![Image](https://github.com/user-attachments/assets/44e87d7b-ac6f-44a1-9cb7-8db97d6bf316)

**Comment by phirsch** [comment]
Thanks a lot! Great to see this (and, more generally, steerable models) included. (I noticed the announcement, but it took me a while to realise I suggested it in the first place... 🙃).

**Comment by phirsch** [comment]
I do have to agree with you though that voice consistency between adjacent sentences still isn't great yet. (Given that consistency _within_ a sentence is high and based on my understanding of the general internal workings of these models, I strongly suspect that it should be possible to carry this latent representation of a voice over from one sentence to the next, but I am not familiar with how to achieve this via the API).

**Comment by mkiol** [comment]
> I strongly suspect that it should be possible to carry this latent representation of a voice over from one sentence to the next, but I am not familiar with how to achieve this via the API).

I have tried many approaches, including using the same "seed" and each time there were differences. From my tests, the best consistency is provided by [Parler-TTS Mini Expresso](https://huggingface.co/parler-tts/parler-tts-mini-expresso) and the use of the names "Jerry, Thomas, Elisabeth, Talia" in the description.

By default, Speech Note splits text into sentences to speed up reading, which makes this problem even more annoying. There is an option to disable splitting, so the text is processed in one pass, which makes the voice consistent, but the generation is much slower :/

![Image](https://github.com/user-attachments/assets/a51d58aa-0c5b-4c59-8622-c9b64adfe31a)

**Comment by mkiol** [comment]
Support for Parler-TTS is included in the new 4.8.0 version.
